File: services/TaskService/Models/Task.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TaskService.Models;

[Table("Tasks", Schema = "Tasks")]
public class Task
{
    public Guid Id { get; set; }

    [Required]
    [StringLength(200)]
    public string Title { get; set; } = string.Empty;

    public string? Description { get; set; }

    [Required]
    [StringLength(50)]
    public string Status { get; set; } = "todo"; // todo, inprogress, done, cancelled

    public Guid? AssigneeId { get; set; } // References UserService user

    [Required]
    public Guid CreatedBy { get; set; } // User who created the task

    public int Priority { get; set; } = 3; // 1=High, 2=Medium, 3=Low

    public DateTime? DueDate { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/task.ts
export interface Task {
  id: string;
  title: string;
  description?: string;
  status: "todo" | "inprogress" | "done" | "cancelled";
  assigneeId?: string;
  createdBy: string;
  priority: 1 | 2 | 3; // 1=High, 2=Medium, 3=Low
  dueDate?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateTaskRequest {
  title: string;
  description?: string;
  assigneeId?: string;
  priority: 1 | 2 | 3;
  dueDate?: string;
}

export interface UpdateTaskRequest {
  title: string;
  description?: string;
  status: string;
  assigneeId?: string;
  priority: 1 | 2 | 3;
  dueDate?: string;
}

export interface AssignTaskRequest {
  assigneeId: string;
}

export interface UpdateTaskStatusRequest {
  status: string;
}

export interface TasksResponse {
  tasks: Task[];
}

export interface TaskResponse {
  task: Task;
}
